Grammar, flow, and spelling corrections

settermjd opened the pull request:
This fixes #3333.

voroyam commented:
@settermjd  can you give me a feedback on the last few comments?

voroyam commented:
also - I have been continuously adding things to the issue, don't know when you worked in all the comments, but I think you have missed the most resent changes

settermjd commented:
@voroyam, I'm guessing you added more after I submitted the PR. I'll have a look now.

settermjd commented:
@voroyam, I've updated the initial commit and added two subsequent ones. Let me know what you think. I don't agree with all the suggestions. However, you made a series of good ones, which I've done my best to implement.

settermjd commented:
@voroyam, ready for a further review.
